integration/e2e/service.go

```diff
@@ -571,6 +571,32 @@ func (s *HTTPService) SumMetrics(metricNames ...string) ([]float64, error) {
 	return sums, nil
 }
 
+// Waits until promQL expression returns scalar (or vector with single value) with non-zero value.
+func (s *HTTPService) WaitForPromQL(expr string) error {
+	lastVal := 0.0
+
+	for s.retryBackoff.Reset(); s.retryBackoff.Ongoing(); {
+		metrics, err := s.Metrics()
+		if err != nil {
+			return err
+		}
+
+		val, err := evaluatePromQLWithTextMetrics(expr, metrics)
+		if err != nil {
+			return err
+		}
+
+		if val > 0 {
+			return nil
+		}
+
+		lastVal = val
+		s.retryBackoff.Wait()
+	}
+
+	return fmt.Errorf("expression %q did not evaluate to positive number, last value: %g", expr, lastVal)
```

Comment:
Last value is always zero right?

Reply:
We now use matchers to compare returned value, so lastValue is useful again.